bazel/pl_build_system.bzl

```diff
@@ -23,8 +23,19 @@ load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library", "cc_test")
 load("@rules_python//python:defs.bzl", "py_test")
 load("//bazel:toolchain_transitions.bzl", "qemu_interactive_runner")
 
+pl_boringcrypto_go_sdk = ["1.20.4"]
 pl_supported_go_sdk_versions = ["1.16", "1.17", "1.18", "1.19", "1.20"]
 
+# The last version in this list corresponds to the boringcrypto go sdk version.
+pl_all_supported_go_sdk_versions = pl_supported_go_sdk_versions + pl_boringcrypto_go_sdk
+
+def pl_go_sdk_version_template_to_label(tpl, version):
+    # If version matches sdk configured to use boringcrypto
+    # the label name should not contain the sdk version string
+    if version == pl_all_supported_go_sdk_versions[-1]:
+        return tpl % "boringcrypto"
+    return tpl % version.replace(".", "_")
```

Comment:
The goal of this function was to make provide stable names for boringcrypto targets. Whenever we upgrade our latest version of go and the boringcrypto version, I wanted to avoid having to change version strings across all the boringcrypto targets.

Reply:
> The goal of this function was to make provide stable names for boringcrypto targets. Whenever we upgrade our latest version of go and the boringcrypto version, I wanted to avoid having to change version strings across all the boringcrypto targets.

This seems fine but also "Whenever we upgrade our latest version of go" we do have to change version strings across all the tests that consume version specific go binaries. So IMO having to change the boringcrypto target deps at the same time seems fine (though I thing having to rev at patch versions could get annoying, but if you dropped the patch version but made it an versioned string, it seems ok)

Reply:
I think I'd prefer to keep it this way for now. If we need to test against boringcrypto enabled binaries across multiple go versions, then I think including the major and minor (similar to how the existing labels work) makes sense.